### PR TITLE
Updates to deps, adding docs, adding optional typescript compiler override.

### PR DESCRIPTION
**Pull request by dzearing**

### CODE REVIEW

**Comment by iclanton** [comment]
Approved.

File: src/TypeScriptTask.ts [hunk 1]
```diff
@@ -73,7 +96,8 @@ export class TypeScriptTask extends GulpTask<ITypeScriptTaskConfig> {
 
     const tsCompilerOptions: ts.Params = assign({}, tsConfig.compilerOptions, {
       module: 'commonjs',
-      sortOutput: true
+      sortOutput: true,
+      typescript: this.taskConfig.typescript
```
Comment (pgonzal) [inline src/TypeScriptTask.ts]: If the typescript object is NOT provided, where does the default come from?

**Comment by pgonzal** [comment]
Approved.

File: src/TypeScriptTask.ts (hunk 1 above)
Comment (iclanton) [inline src/TypeScriptTask.ts]: If the `taskConfig.typescript` value is `undefined`, it'll behave the same way it did before this PR and the TS compiler will construct one automatically.

**Pull request merged**